glam/api/views.py

```diff
@@ -155,11 +101,79 @@ def aggregations(request):
 
         record[sub_key]["data"] = data
         response[key] = record
+    return response
+
 
+@api_view(["POST"])
+def aggregations(request):
+    """
+    Fetches aggregation data.
+
+    Expects a JSON object in the body containing the query parameters, e.g.::
+
+        {
+            "query": {
+                "channel": "nightly",
+                "probe": "gc_ms",
+                "versions": ["70"],  # OR ["70", "69", "68"]
+                "aggregationLevel": "version"  # OR "build_id"
+            }
+        }
+
+    Returns a JSON object containing the histogram data and metadata, e.g.::
+
+        {
+            "response": [
+                {
+                    "data": [
+                        {
+                            "client_agg_type": "summed-histogram",
+                            "histogram": {
+                                "0": 0.0,
+                                "1": 1920.963,
+                                ...
+                            },
+                            "percentiles": {
+                                "0": 1.0,
+                                "10": 1.0259,
+                                ...
+                            },
+                            "total_users": 1604
+                        }
+                    ],
+                    "metadata": {
+                        "build_id": null,
+                        "channel": "nightly",
+                        "metric": "gc_ms",
+                        "metric_type": "histogram-exponential",
+                        "os": "Linux",
+                        "version": "70"
+                    }
+                }
+            ]
+        }
+
+    """
+
+    body = request.data
+
+    if body is None or body.get("query") is None:
+        raise ValidationError("Unexpected JSON body")
+
+    q = body["query"]
+
+    response = get_aggregations(
```

Comment:
You can probably just replace this with:

    response = get_aggregations(body["query"])

Since `get_aggregations` expects a dict, and then let it figure out if the required params are there.

Reply:
👍